src/parquet/schema/descriptor.cc

```diff
@@ -81,9 +82,19 @@ ColumnDescriptor::ColumnDescriptor(const schema::NodePtr& node,
 }
 
 const ColumnDescriptor* SchemaDescriptor::Column(int i) const {
+  if (i < 0 || i >= static_cast<int>(leaves_.size())) {
+    throw ParquetException("Invalid column id " + std::to_string(i));
+  }
   return &leaves_[i];
 }
 
+const schema::NodePtr& SchemaDescriptor::base(int i) const {
+  if (i < 0 || i >= static_cast<int>(leaves_.size())) {
+    throw ParquetException("Invalid column id " + std::to_string(i));
+  }
```

Comment:
I'm a bit -0 on boundschecking / throwing an exception in these cases. We could use `DCHECK` instead since these would indicate a bug in client code rather than an exception that you would want to catch. 

@xhochy do you have any opinion on this?